src/Move/Address.h

```diff
@@ -50,7 +57,7 @@ class Address {
             throw std::invalid_argument("Invalid address string");
         }
         auto hexFunctor = [&string]() {
-            if (std::size_t hexLen = string.size() - 2; string.starts_with("0x") && hexLen < size * 2) {
+            if (std::size_t hexLen = string.size() - 2; string.starts_with("0x") && hexLen < hexSizeAddress && (string.size() == shortSizeAddress || hexLen == hexNonPaddedSizeAddress)) {
```

Comment:
Please consider the following:
```c++
std::size_t hexLen = string.size() - 2;
bool isExpectedLen = hexLen == hexShortSizeAddress || hexLen == hexNonPaddedSizeAddress;
if (string.starts_with("0x") && isExpectedLen) {
  // ...
}
```

`hexLen < hexSizeAddress` is not needed as we expect the `hexLen` to be either `hexShortSizeAddress` or `hexNonPaddedSizeAddress`.